src/main/java/dk/kb/storage/facade/DsStorageFacade.java

```diff
@@ -102,14 +104,65 @@ public static DsRecordDto getRecord(String recordId) {
                 return null;
             }
             
-            if (record.getParentId() == null) { //can not have children if also has parent (1 level only hieracy)
-                ArrayList<String> childrenIds = storage.getChildrenIds(record.getId());
-                record.setChildrenIds(childrenIds);
-            }
+
+           ArrayList<String> childrenIds = storage.getChildrenIds(record.getId());
+           record.setChildrenIds(childrenIds);
+
             return record;
         });
     }
 
+    /**
+     *   Will load full object tree. The DsDecordDto return will a pointer the record with the recordId in the tree
+     *       
+     *  Logic: Find top parent recursive and load children.
+     * 
+     *  @param recordId The full object tree will be returned with a pointer to this record   
+     * 
+     *  Return null if record does not exist.
+     */
+    public static DsRecordDto getRecordTree(String recordId) {
+      
+        
+        return performStorageAction("getRecord(" + recordId + ")", storage -> {
+        String idNorm = IdNormaliser.normaliseId(recordId);          
+        DsRecordDto record = getRecord(idNorm); //Load from facade as this will set children
+        
+        if (record== null) {
+            return record;
+        }
+         DsRecordDto topParent = getTopParent(record); //this will also detect a cycle.              
+         
+         List<DsRecordDto> xList = new ArrayList<DsRecordDto>();
+         
+         loadAndSetChildRelations(topParent,null, recordId, xList); //Resursive method     
+                   
+         //Sanity check, just in case..
+          if (xList.size() !=1) {
+              log.error("Logic error loading recordTree with id:"+recordId);
+              throw new InternalServiceException("Logic error loading recordTree with id:"+recordId);
+          }
+          
+         return xList.get(0);
+         
+        });
+    }
+    
+    
+    //Will follow parent recursive until there no longer exists a parent. (root of object tree)
+    private static DsRecordDto getTopParent(DsRecordDto record) {
+      DsRecordDto topParent = record;
+      while (topParent.getParentId() != null) {          
+          if (topParent.getParentId().equals(topParent.getId())) {
+              log.error("Invalid record, has itself as parent, id:"+topParent.getId());
+              return topParent; // Throw exception instead?
```

Comment:
For a singular `getRecordTree`then yes. If we expand this so that `getRecords` can also expand records, we might want to reconsider.